client/index.js

```diff
@@ -9,27 +9,27 @@ module.exports = class Client {
   async getPublications (options) {
```

Comment:
maybe also get rid of the `async` not necessary as well

Reply:
Would you also remove for `getDesignVersions` where we have assertions?
I would, but I'm not sure how people would expect the error behavior.
e.g. if you manually call `.catch`, the errors won't show up in there because it fails before the promise is constructed.

Reply:
Maybe it would be better to explicitly `return Promise.resolve(error)` instead of throwing errors? What do you think?

Reply:
Yeah both ways have their pros/cons :) I am open to both actually!

Reply:
I just tested with `await`
```
function foobar () {
  throw new Error('Error in foobar')
}

function promfoobar () {
  return Promise.reject(new Error('Error in foobar'))
}

async function asyncfoobar () {
  throw new Error('Error in foobar')
}

async function asyncpromfoobar () {
  return Promise.reject(new Error('Error in foobar'))
}
```

And then with `await functionname()`... all result in the same stack trace. So it only affects the manual `.then(...).catch(...)` calls.

Reply:
Let's keep this for now and let's test in a real app.

Reply:
Lets use the third:

```js
async function asyncfoobar () {
  throw new Error('Error in foobar')
}
```

After the whole discussion this convinces me the most. It makes clear that this is async / a promise and should support the `then().catch()` case 👍

Reply:
Thanks for going the extra mile and investigate this with me! 🎉

Reply:
Maybe this could be something @mokimo could test in a delivery.

Reply:
> Maybe it would be better to explicitly return Promise.reject(error) instead of throwing errors? > What do you think?

Browsers don't properly support async/await, so in some cases we might want to use `Promise.reject`. Let's see.